Tests/Marketplace/marketplace_services.py

```diff
@@ -1567,6 +1651,179 @@ def prepare_release_notes(self, index_folder_path, build_number,
+                                                     return_str=False)
+        if self.is_release_notes_not_for_entities(release_notes_str=release_notes, release_notes_dict=release_notes_dict):
+            return changelog_entry, False
+
+        filtered_release_notes_from_tags = self.filter_headers_without_entries(release_notes_dict)  # type: ignore[arg-type]
+        filtered_release_notes = self.filter_release_notes_by_entities_display_name(filtered_release_notes_from_tags,
+                                                                                    modified_files_data)
+
+        if modified_files_data and not filtered_release_notes:
+            return {}, True
+
+        # Convert the RN dict to string
+        changelog_entry[Changelog.RELEASE_NOTES] = construct_entities_block(filtered_release_notes).strip()
+        return changelog_entry, False
+
+    def filter_release_notes_by_entities_display_name(self, release_notes, modified_files):
+        """
+        Filters the RN entries by the modified files display names given from id-set.
+
+        Args:
+            release_notes (dict): The release notes in a dict object.
+            modified_files (dict): The modified files data that are given from id-set.
+
+        Return:
+            (dict) The filtered release notes.
+        """
+        filtered_release_notes: dict = {}
+        for pack_folder, entities_data in modified_files.items():
+
+            rn_header = RN_HEADER_BY_PACK_FOLDER[pack_folder]
+
+            # This might be if the entity was filtered by the tags
+            if rn_header not in release_notes:
+                continue
+
+            # Filters the RN entries by the entity display name
+            display_names = [entity['display_name'] for entity in entities_data]
+            filtered_entries = self.filter_special_entities(release_notes, display_names, rn_header)
+
+            if filtered_entries:
+                logging.info(f"Found relevant entries after filtering by display_name - {filtered_entries}")
+                filtered_release_notes[rn_header] = filtered_entries
+
+        return filtered_release_notes
+
+    @staticmethod
+    def filter_special_entities(release_notes: dict, display_names: list, rn_header: str):
+        """
+        Filters the entries by display names and also handles special entities that their display name is not an header.
+
+        Args:
+            release_notes (dict): The release notes in a dict.
+            display_names (list): The display names that are give from the id-set.
+            rn_header (str): The release notes entity header.
+
+        Returns:
+            (dict) The filtered release notes entries.
+        """
+        filtered_entries: dict = {}
+        for display_name, rn_entry in release_notes[rn_header].items():
+
+            if display_name != '[special_msg]' and display_name.replace("New: ", "") not in display_names:
+                continue
```

Comment:
I understand that here occurs the filtering of entries whose display name is not in the id set. If so, this function name cannot be named `filter_special_entities` because it does something extra. Try to think of a different name or even change the order of the logic a bit.